CMakeLists.txt

```diff
@@ -429,6 +430,7 @@ if(BUILD_TEST)
     ${NVFUSER_ROOT}/test/test_gpu_indexing_ops.cpp
     ${NVFUSER_ROOT}/test/test_gpu_indexing.cpp
     ${NVFUSER_ROOT}/test/test_gpu_multidevice.cpp
+    ${NVFUSER_ROOT}/test/test_gpu_multidevice_collectives.cpp
```

Comment:
nit: no need to have "gpu" anymore, so `test_multidevice_collectives.cpp` is just fine